Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1026,6 +1022,133 @@ namespace AZ
             rayTracingFeatureProcessor->SetMesh(m_objectId, m_model->GetModelAsset()->GetId(), subMeshes);
         }
 
+        void ModelDataInstance::SetIrradianceData(
+            RayTracingFeatureProcessor::SubMesh& subMesh,
+            const Data::Instance<RPI::Material> material,
+            const Data::Instance<RPI::Image> baseColorImage)
+        {
+            RPI::MaterialPropertyIndex propertyIndex;
+
+            // Default to manual color, in case the material type doesn't have the concept of
+            // manual vs. automatic irradiance color.
+            AZ::Name irradianceColorSource = AZ::Name("Manual");
+            propertyIndex = material->FindPropertyIndex(AZ::Name("irradiance.irradianceColorSource"));
+            if (propertyIndex.IsValid())
+            {
+                uint32_t enumVal = material->GetPropertyValue<uint32_t>(propertyIndex);
+                irradianceColorSource = material->GetMaterialPropertiesLayout()
+                                                ->GetPropertyDescriptor(propertyIndex)
+                                                ->GetEnumName(enumVal);
+            }
+            if (irradianceColorSource == AZ::Name("BaseColorTint"))
+            {
+                // Use only the baseColor, no texture on top of it
+                subMesh.m_irradianceColor = subMesh.m_baseColor;
+            }
+            if (irradianceColorSource == AZ::Name("BaseColor"))
+            {
```

Comment:
Just to make sure I understand this correctly, those AZ::Names should for instance be stored as member variables in the ModelDataInstance object (e.g. in the style of 'm_nameBaseColorTint', 'm_nameBaseColor', possibly something like 'm_nameBaseColor_textureBlendMode' for 'baseColor.textureBlendMod'?) such that those strings are only hashed once and stay in the name dictionary for the lifetime of the ModelDataInstance object?

Note: this would amount to around 20 such members when applied to all occurrences of AZ::Names within the MeshFeatureProcessor.cpp source file (25 when including the names used in constructing the RHI::ShaderSemantic objects) which might get a bit unwieldy -- unless there is a cleaner alternative that I'm missing?

Reply:
Yes, I believe that's what @rgba16f is suggesting. But in this case I don't agree with making that change. There is an effort to add support for statically initialized Name objects, which will eventually allow us to keep these Names in place and just make them local static const variables. It might not be worth the effort to make a bunch of Name members, as changing them to statics in a few months (hopefully) would be an easy change. Also, I worry that it would be premature optimization; this code is mostly run at build time or maybe rare occasions at load time.